### DISCUSSION TITLE
PlugData v0.5.2

### DISCUSSION THREAD

**Discussion by timothyschoen** [plugdata-team/plugdata#91] [Announcements]
Version with important fixes for problems in the last version.

- Fixed crash when double clicking folder in document browser (thanks @jaffasplaffa for reporting)
- Fixed crash on Windows when closing plugin editor (thanks @offthesky and @alfonso73 for reporting)
- Fixed standalone very slow on Windows (thanks @charlesneimog for reporting)

<hr /><em>This discussion was created from the release <a href='https://github.com/timothyschoen/PlugData/releases/tag/v0.5.2'>PlugData v0.5.2</a>.</em>

**Comment by ludnny** [comment]
Thank you for this fast fix-version!

All the critical problems from the previous version 0.5.1 Standalone on Windows are fixed here (super slow responsiveness / fullscreen app was not totally fullscreen because of shadows I guess, starting time was super slow). 

At first glance everything is working fine. 

Tested as well with Reaper VST3, with parameter automation: works magically.

Congrats again!

**Comment by alfonso73** [comment]
The Windows GUI problems seems solved...i still have several and frequent crashes in Reaper (64x) with PlugData VST3. I'm often using ELSE stuff and did some test with [bl.vsaw~] that's actually an ELSE oversampling abstraction containing the ELSE object [vsaw~]. Probably there's is something goin' on...i'll test with pure PD objects to see if i have frequent crashes to.

**Comment by JoshuaACNewman** [comment]
Very excited about this project. I'm a makerspace teacher, looking to use Pd to teach programming and the clear UI is going to do enormous work encouraging experimentation from the kids.

**Reply by timothyschoen** [reply]
That so cool! Max and Pd also got me into programming, it's much more encouraging when you're just starting out. This is a use case that I tried to keep in mind, when PlugData is done it should have inlet/outlet hover message for all objects, which should really help anyone who's new to Pd. If you have any other ideas that will make it easier for beginners, let me know!

**Reply by JoshuaACNewman** [reply]
Absolutely! Maybe I’ll start an education/accessibility ticket to discuss.

I should note, my abilities with lexical programming are pretty poor. I’ve done a lot of programming, particularly for microcontrollers, but it’s a struggle. Pd is something I’ve always wished was better in exactly the ways you’re improving it.

**Reply by timothyschoen** [reply]
> Absolutely! Maybe I’ll start an education/accessibility ticket to discuss.
> 
> I should note, my abilities with lexical programming are pretty poor. I’ve done a lot of programming, particularly for microcontrollers, but it’s a struggle. Pd is something I’ve always wished was better in exactly the ways you’re improving it.

Feel free to open a ticket, accessibility suggestions are always welcome!

**Comment by 60-hz** [comment]
Of course inlet/outlet hover messages are very convenient for learning that kind of software.
Why not poping object's basic function while hovering, because user might not guess what a [else/bl.vsaw~] is all about at first sight and many objet's name in pure data libs are cranky.
That's why I am using ceammc library during class: objets' name are much more understandable.

**Reply by timothyschoen** [reply]
I agree, and it should be simple to do as well. I'm already collecting descriptions for the auto-complete menu anyways. Most of the work now is actually copying and checking descriptions from help files. I've been putting it off for a while, but since I just graduated I'll have some more time for it soon!

**Comment by JoshuaACNewman** [comment]
A complete menu/palette that allows you to know what’s available could be
extremely valuable, not just for new users, but also for when you’re trying
out something new and don’t know whether to look for attenuator~ or vca~ or
whatever.

VCV does this by categorizing them. I don’t know if there’s any metadata
for objects that’s like that.

⊦⪽

On Thu, Jun 23, 2022 at 11:20 AM Timothy Schoen ***@***.***>
wrote:

> I agree, and it should be simple to do as well. I'm already collecting
> descriptions for the auto-complete menu anyways. Most of the work now is
> actually copying and checking descriptions from help files. I've been
> putting it off for a while, but since I just graduated I'll have some more
> time for it soon!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timothyschoen/PlugData/discussions/91#discussioncomment-3011316>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGUUIQ57VWSKA2Y5WRKJWP3VQR6CTANCNFSM5VPSRYDQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
*⊦⪽ *

**Reply by timothyschoen** [reply]
The biggest problem here is indeed the metadata. PlugData has quite a few objects because it includes ELSE, so writing metadata for everything will be a big task. I agree that some kind over overview would be nice though, I'll think about a nice way to this!

**Comment by JoshuaACNewman** [comment]
This feels like a community project. How many objects are there in ELSE?

But also, if you start with the metadata for everything that’s on a
semimodular, (oscillators, filters, VCAs, ADC and DAC) plus basic logic,
that will probably answer the questions a newcomer would have to give them
the tools to ask questions and build help objects for them.

On Thu, Jun 23, 2022 at 3:37 PM Timothy Schoen ***@***.***>
wrote:

> The biggest problem here is indeed the metadata. PlugData has quite a few
> objects because it includes ELSE, so writing metadata for everything will
> be a big task. I agree that some kind over overview would be nice though,
> I'll think about a nice way to this!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timothyschoen/PlugData/discussions/91#discussioncomment-3013065>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGUUIQYVCAPWVPXOANTH5YTVQS4GJANCNFSM5VPSRYDQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
*⊦⪽ *

**Reply by timothyschoen** [reply]
The community project was once started [here]( https://github.com/timothyschoen/PlugData/issues/75), but it got interrupted because I found a much simpler format for it. Current status is that vanilla Pd objects have documentation, but it's sometimes incomplete or too verbose. ELSE and cyclone still need to be documented, ELSE has 456 objects, cyclone has around 200.

And we could create a basic panel/cheat sheet for the most important objects, that should work
